### PR TITLE
Bug/wa-55 Jupyter app launched natively on HPC system

### PR DESCRIPTION
**Pull request by chandra-tacc**
### Context
[WA-55](https://jira.tacc.utexas.edu/browse/WA-55)

### Fix
* Use cmdPrefix option in tapis and run the jupyter app in login node.
* The container is a no-op. This app uses debian and simply send an echo and exits.
* The script used in cmdPrefix is in corral. Due to bugs [WP-314](https://jira.tacc.utexas.edu/browse/WP-314) and [WP-313](https://jira.tacc.utexas.edu/browse/WP-313), the path to the script is directly referenced in cmdPrefix instead of using tapis input and copying locally to execution folder.
* Jupyter notebook is the only option available in Frontera

### Testing

Validated on dev.cep. Screenshots below

App:
Screenshot 2023-10-06 at 8 27 48 AM

Connect button available:
Screenshot 2023-10-06 at 8 29 56 AM

Jupyter Notebook:

Screenshot 2023-10-06 at 9 05 59 AM

Job finished after session file is deleted:
Screenshot 2023-10-06 at 9 10 42 AM

Snippet of tapis output file:

```TACC: Your jupyter notebook server is now running at https://frontera.tacc.utexas.edu:60079/?token=29b5e54c35dba96dc3561685793422a800e64412d165e8c6a7bd908baea46417
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   224  100     2  100   222      5    630 --:--:-- --:--:-- --:--:--   630
OKTACC: release port returned {"released_port": 60079}
INFO:    Using cached SIF image
Job is done.```

### CODE REVIEW

File: applications/jupyter-hpc-mpi/app.json [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        "maxMinutes": 120,
+        "subscriptions": [],
+        "tags": []
+    },
+    "tags": [
+        "portalName: cep"
+    ],
+    "notes": {
+        "label": "Jupyter HPC MPI (Frontera)",
+        "helpUrl": "https://jupyterlab.readthedocs.io/en/stable/",
```
Comment (rstijerina) [inline applications/jupyter-hpc-mpi/app.json:83]:
```suggestion
        "helpUrl": "https://jupyter-notebook.readthedocs.io/en/stable/",
```

File: applications/jupyter-hpc-mpi/app.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    "version": "1.0.1",
+    "description": "Run an interactive Jupyter session with ability to launch mpi jobs.",
+    "owner": "${apiUserId}",
+    "enabled": true,
+    "runtime": "SINGULARITY",
+    "runtimeVersion": null,
+    "runtimeOptions": [
+        "SINGULARITY_RUN"
+    ],
+    "containerImage": "docker://debian:bullseye-slim",
```
Comment (rstijerina) [inline applications/jupyter-hpc-mpi/app.json:12]: Could use `docker://alpine:3` for an even smaller no-op image

Comment (chandra-tacc) [reply applications/jupyter-hpc-mpi/app.json:12]: > Could use `docker://alpine:3` for an even smaller no-op image

Yes, I started with alpine. But, alpine fails to run with this error:
```
INFO:    Creating SIF file...
Error relocating /opt/apps/xalt/xalt/lib64/libxalt_init.so: __strdup: symbol not found
```
interestingly, when I use idev, it runs fine. There are no bind mounts in the app definition. I'm assuming some libs are bound automatically when running via the slurm job. No, error with debian, so choose the safer option.

Comment (rstijerina) [reply applications/jupyter-hpc-mpi/app.json:12]: Interesting, good choice 👍

**Pull request merged**